### PR TITLE
Provide expiry and bufferLength configuration properties for configuring a distribution statistic

### PR DESCRIPTION
**Pull request by polarbear567**
To enhance the [27477](https://github.com/spring-projects/spring-boot/issues/27477).

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java [hunk 1]
```diff
@@ -296,6 +297,10 @@ public AutoTimeProperties getAutotime() {
 		 */
 		private final Map<String, String> maximumExpectedValue = new LinkedHashMap<>();
 
+		private final Map<String, Duration> expiry = new LinkedHashMap<>();
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java:300]: This should have a description like the other fields.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java [hunk 2]
```diff
@@ -296,6 +297,10 @@ public AutoTimeProperties getAutotime() {
...
+
+		private final Map<String, Integer> bufferLength = new LinkedHashMap<>();
```
Comment (snicoll) [inline spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java:302]: There's a description missing here as well.

**Review by snicoll — CHANGES_REQUESTED** [review]
Thanks for the PR. Before we consider this change we would need additional tests that exercise the code you've added. Can you please review?

**Comment by polarbear567** [comment]
> Thanks for the PR. Before we consider this change we would need additional tests that exercise the code you've added. Can you please review?

No problem. I'm worried that my comments can't be accurately expressed. Can you review the comments?

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java (hunk 1 above)
Comment (polarbear567) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java:305]: Please review the description.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java (hunk 2 above)
Comment (polarbear567) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsProperties.java:313]: Please review the description.

**Comment by snicoll** [comment]
No worries, I can polish as part of the merge. Can you please add some tests?

**Comment by polarbear567** [comment]
> No worries, I can polish as part of the merge. Can you please add some tests?

Sure. I'll add some tests in PropertiesMeterFilterTests.

**Comment by polarbear567** [comment]
some tests have been added, please review.

**Pull request closed without merging**